core/src/main/java/org/apache/calcite/runtime/SqlFunctions.java

```diff
@@ -1854,6 +1854,21 @@ public static double acos(double b0) {
     return Math.acos(b0);
   }
 
+  // ACOSH
+  /** SQL <code>ACOSH</code> operator applied to BigDecimal values. */
+  public static double acosh(BigDecimal b0) {
+    return acosh(b0.doubleValue());
+  }
+
+  /** SQL <code>ACOSH</code> operator applied to double values. */
+  public static double acosh(double b0) {
+    if (b0 < 1) {
+      SQLException e = new SQLException("Input parameter of acosh cannot be small than 1!");
```

Comment:
I would lean towards throwing an IllegalArgumentException in this case (same applies to the `atanh`method).
Also, I think it would me more correct if the message says "... cannot be less than 1".

Reply:
Thanks, I will update it soon